chain/chain/src/store/mod.rs

```diff
@@ -498,32 +498,20 @@ impl ChainStore {
 
     pub fn compute_transaction_validity(
         &self,
-        protocol_version: ProtocolVersion,
         prev_block_header: &BlockHeader,
         chunk: &ShardChunk,
-    ) -> Result<Vec<bool>, Error> {
-        let relaxed_chunk_validation =
-            ProtocolFeature::RelaxedChunkValidation.enabled(protocol_version);
-        let mut valid_txs = Vec::with_capacity(chunk.transactions().len());
-        if relaxed_chunk_validation {
-            for transaction in chunk.transactions() {
-                let is_valid = self.check_transaction_validity_period(
-                    prev_block_header,
-                    transaction.transaction.block_hash(),
-                );
-                valid_txs.push(is_valid.is_ok());
-            }
-        } else {
-            for transaction in chunk.transactions() {
+    ) -> Vec<bool> {
+        chunk
+            .transactions()
+            .into_iter()
```

Comment:
personally I'm more of a for loop guy but I guess it's personal preference

Reply:
I know what you mean.  I like this approach mainly because there are fewer mutable variables and less "cruft" i.e. declaring empty Vec, pushing to it, etc. etc.